engine/src/main/java/org/terasology/world/block/BlockRegionIterable.java

```
@@ -0,0 +1,134 @@
+                    for (BlockRegion blockRegion : subtract) {
+                        if (blockRegion.testBlock(test)) {
+                            return false;
+                        }
+                    }
+                }
+                return true;
+            }
+
+            public boolean findNext() {
+                if (current.equals(next)) {
+                    do {
+                        next.z++;
+                        if (next.z >= region.maxZ) {
+                            next.z = region.minZ;
+                            next.y++;
+                            if (next.y >= region.maxY) {
+                                next.y = region.minY;
+                                next.x++;
+                            }
+                        }
+                        if (!region.testBlock(next)) {
+                            return false;
+                        }
+                    } while (!isValid(next));
+                    return true;
+                }
+                return true;
+            }
+
+            @Override
+            public boolean hasNext() {
+                if (current == null) {
+                    return true;
+                }
+
+                if (current.equals(next)) {
+                    return findNext();
+                }
+                return !region.testBlock(next);
+            }
+
+            @Override
+            public Vector3ic next() {
+                if (current == null) {
+                    current = new Vector3i(next);
+                    return next;
+                }
+
+                if (current.equals(next)) {
+                    if (findNext()) {
+                        return next;
+                    }
+                    return null;
+                }
+                current.set(next);
+                return next;
+            }
+        };
+    }
+
+    @Override
+    public void forEach(Consumer<? super Vector3ic> consumer) {
+
```

Purposefully empty? Wouldn't it be better to throw something like "UnsupportedOperationException"?